[K9VULN-13941] docs(agentless): document azure cloud shell (#36392)
DataDog/documentation: content/en/security/cloud_security_management/setup/agentless_scanning/enable.md

**Required Azure permissions:** The identity you use in Cloud Shell must have **Owner** (or a role granting role-assignment write and resource creation) on the scanner subscription. On each scanned subscription, you also need a role granting `Microsoft.Authorization/roleAssignments/write` (for example, **User Access Administrator** or **Owner**) so the scanner's managed identity can be granted the permissions it needs to snapshot and read disks. The script creates an Azure Storage Account in the scanner subscription for Terraform state. Alternatively, you can **reuse an existing storage account** by setting the `TF_STATE_STORAGE_ACCOUNT` environment variable to its name; the script will not create one in that case. The required resource providers (`Microsoft.Compute`, `Microsoft.Network`, `Microsoft.ManagedIdentity`, `Microsoft.Storage`, `Microsoft.KeyVault`, `Microsoft.Authorization`) are auto-registered in the scanner subscription when possible.

> 

Reviewer: ```suggestion
1. Ensure you have the required Azure permissions:
   - The identity you use in Cloud Shell must have **Owner** (or a role granting role-assignment write and resource creation) on the scanner subscription.
   - On each scanned subscription, you also need a role granting `Microsoft.Authorization/roleAssignments/write` (for example, **User Access Administrator** or **Owner**) so the scanner's managed identity can be granted the permissions it needs to snapshot and read disks. The script creates an Azure Storage Account in the scanner subscription for Terraform state.
   - Alternatively, you can **reuse an existing storage account** by setting the `TF_STATE_STORAGE_ACCOUNT` environment variable to its name; the script will not create one in that case. The required resource providers (`Microsoft.Compute`, `Microsoft.Network`, `Microsoft.ManagedIdentity`, `Microsoft.Storage`, `Microsoft.KeyVault`, `Microsoft.Authorization`) are auto-registered in the scanner subscription when possible.
```
I'm not sure if I broke these up properly, but listing it as a step and breaking out the separate permissions can help it seem like a little less like a wall of text.

1. On the [Cloud Security Setup](https://app.datadoghq.com/security/configuration/csm/setup) page, click **Cloud Integrations** > **Azure**.
1. Expand the Tenant containing the subscription where you want to deploy the Agentless scanner.
1. Click the **Enable** button for the Azure subscription where you want to deploy the Agentless scanner. The **Vulnerability Scanning** modal opens.

> 1. In the **How would you like to set up Agentless Scanning?** section, select **Cloud Shell**.

Reviewer: ```suggestion
1. Click the **Enable** button for the Azure subscription where you want to deploy the Agentless scanner. The **Vulnerability Scanning** modal opens.
1. Turn on the **Vulnerability Scanning** toggle.
1. In the **How would you like to set up Agentless Scanning?** section, select **Azure Cloud Shell**.
```

1. Select an **API key** that has [Remote Configuration](/remote_configuration) enabled. An application key is automatically generated.
1. Select the **Azure subscriptions** you want to scan.

> 1. Configure the scanner:

Reviewer: This seems to assume that the user clicked "Deploy a new scanner," but skips that click, and doesn't detail what happens if the user takes the recommended route. I'm not sure why that is?

<img width="552" height="145" alt="Image" src="https://github.com/user-attachments/assets/5b43a047-f770-4589-b058-5e79fe3b3612" />

Author: Good call! Reusing existing scanners is a feature that came up later, and I forgot to document it
I've fixed the documentation for GCP as well

   - Select the **scanner subscription** (which must be one of the selected subscriptions).

>    - Select the **scanner locations** (Azure regions, up to four). We recommend deploying scanners in every region where you have more than 150 hosts.

Reviewer: ```suggestion
   - Select the **scanner locations** (Azure regions, up to four). Datadog recommends deploying scanners in every region where you have more than 150 hosts.
```

<div class="alert alert-info">If you haven't connected your Azure subscription to Datadog yet, <a href="https://app.datadoghq.com/security/configuration/csm/setup?active_steps=cloud-accounts&active_sub_step=azure">set up the Azure integration</a> first.</div>

{{% collapse-content title="Cloud Shell" level="h3" id="azure-cloud-shell-setup" %}}
Use Azure Cloud Shell to set up Agentless Scanning for your Azure subscriptions. This method downloads a [setup script](https://github.com/DataDog/integrations-management/tree/main/azure/agentless) that wraps the [Terraform Datadog Agentless Scanner module for Azure](https://github.com/DataDog/terraform-module-datadog-agentless-scanner/tree/main/azure#readme), so you do not need to manage Terraform directly. You can review the script before running it.

1. Ensure you have the required Azure permissions:

   - On the **scanner subscription**, the identity you use in Cloud Shell must have **Owner** (or a role granting role-assignment write and resource creation).

>    - On **each scanned subscription**, you need a role granting `Microsoft.Authorization/roleAssignments/write` (for example, **User Access Administrator** or **Owner**) so the scanner's managed identity can be granted the permissions it needs to snapshot and read disks.

Reviewer: ```suggestion
1. Ensure the identity you use in Cloud Shell has the required Azure permissions:

   - On the **scanner subscription**, the identity must have a role that grants role-assignment write and resource creation, such as **Owner**.
   - On **each scanned subscription**, the identity must have a role that grants the `Microsoft.Authorization/roleAssignments/write` permission, so the scanner's managed identity can be granted the permissions it needs to snapshot and read disks, such as **User Access Administrator** or **Owner**.
```

   - For Terraform state, the script creates an Azure Storage Account in the **scanner** subscription unless you set `TF_STATE_STORAGE_ACCOUNT` to reuse an existing storage account in the scanner resource group; the script does not create a new account in that case.

>    - The script registers these resource providers in the scanner subscription when possible: `Microsoft.Compute`, `Microsoft.Network`, `Microsoft.ManagedIdentity`, `Microsoft.Storage`, `Microsoft.KeyVault`, `Microsoft.Authorization`.

Reviewer: These aren't permissions, so they don't make sense under the "ensure you have the required Azure permissions:" list.

I'm trying to figure out if this information is important in the context of this step in the procedure. If they're important for a user to address before proceeding, maybe we can make the action items more clear. If they're not important for this particular step, maybe there's a more appropriate place we can move them to.

Author: Thanks for calling that out. I moved Terraform state and resource provider registration out of the “required Azure permissions” list into a short “Before you run the generated command…” section so that list is only IAM. Happy to tweak wording if you want that block titled differently.

1. On the [Cloud Security Setup](https://app.datadoghq.com/security/configuration/csm/setup) page, click **Cloud Integrations** > **Azure**.

> 1. Expand the Tenant containing the subscription where you want to deploy the Agentless scanner.

Reviewer: Double checking whether this is still a thing? When I look in the app, it looks like everything is automatically expanded?

Author: Good catch. The UI often shows everything expanded already. I changed the step to navigate under the tenant that contains the subscription and click Enable, and added a line to expand the tenant or subscription list only if it’s collapsed. I used the same pattern for GCP (account / project).

> 1. Click the **Enable** button for the Azure subscription where you want to deploy the Agentless scanner. The **Vulnerability Scanning** modal opens.

Reviewer: ```suggestion
1. Click **Enable** for the Azure subscription where you want to deploy the Agentless scanner. The **Vulnerability Scanning** modal opens.
```
Very picky, but usually we skip the work of telling the user it's a button because they can see that 🙂

{{% collapse-content title="Cloud Shell" level="h3" id="gcp-cloud-shell-setup" %}}
Use Google Cloud Shell to set up Agentless Scanning for your GCP projects. This method downloads a [setup script](https://github.com/DataDog/integrations-management/tree/main/gcp/agentless) that wraps the [Terraform Datadog Agentless Scanner module for GCP](https://github.com/DataDog/terraform-module-datadog-agentless-scanner/tree/main/gcp#readme), so you do not need to manage Terraform directly. You can review the script before running it.

1. Ensure you have the required GCP permissions:

   - On the **scanner project**, the identity you use in Cloud Shell must have **Owner** or equivalent.
   - The script creates a GCS bucket for Terraform state in that project, so you also need **Storage** permissions (for example, `roles/storage.admin` or `storage.buckets.create` / `storage.buckets.get` / `storage.buckets.update`). Alternatively, set `TF_STATE_BUCKET` to reuse an existing bucket; the script does not create a bucket in that case.

>    - If you see a 403 error on "Setting up Terraform state storage", see [GCP: Failed to create state bucket][26] in the troubleshooting guide.

Reviewer: Same edits as above - both wording, and making sure that if we introduce a list as "required GCP permissions," that all items in that list relate to those permissions. Let me know if you'd like some help moving stuff around 🙂

Author: Same treatment as Azure: the numbered prerequisites list is now strictly permissions (Owner/equivalent plus Storage for state, including reuse via TF_STATE_BUCKET), and default bucket behavior plus the 403 troubleshooting link live in the “Before you run the generated command…” section so the list stays consistent. Let me know if you’d like the Storage bullet tightened further.
